### DISCUSSION TITLE
getListingsByShop tell me I'm missing parameter "[quantity]", but it doesn't exist.

### DISCUSSION THREAD

**Discussion by ClientConvert** [etsy/open-api#1120] [Q&A]
I'm trying to get a list of all listings in my store.
When I use the end point
https://openapi.etsy.com/v3/application/shops/{shop_id}/listings
as seen in the docs here (https://developer.etsy.com/documentation/reference/#operation/getListingsByShop)
I get the response 400
`{"error":"Missing input parameter: [quantity]"}`
However when you look in the docs, quantity is not valid parameter.

My big question is.. How do I get a list of all store items regardless of state (active, draft...)?

**Comment by seandmccarthy** [comment]
> My big question is.. How do I get a list of all store items regardless of state (active, draft...)?

You have to ask for the listings in each state in turn. You'll also need to "page" through the results using `offset` as you can only request a maximum of 100 listings at a time (better hope that listings aren't being inserted while you're trying to iterate through the list...).

We loop over each state that we want to read (e.g. `["active", "inactive", "sold_out", "draft", "expired"]`), then iterate within each state calling `getListingsByShop` with the state and incrementing the offset until we get zero results back.

I don't know why you're being asked for quanity, but our typical request looks like:

`https://api.etsy.com/v3/application/shops/{shop_id}/listings?state=active&sort_on=created&sort_order=asc&includes=Images%2CInventory%2CShipping%2CVideos&limit=100&offset=0`
